tests/plugins/contextmenu/keystroke.js

```diff
@@ -0,0 +1,55 @@
+/* bender-tags: editor, bug */
+/* bender-ckeditor-plugins: clipboard,contextmenu */
+
+( function() {
+	'use strict';
+
+	bender.editors = {
+		classic: {
+			name: 'classic'
+		},
+		inline: {
+			name: 'inline',
+			creator: 'inline'
+		}
+	};
+
+	var tests = {
+		'Opening context menu with keystroke': function( editor ) {
+			// Test is unstable on Firefox, because context menu is showing asynchronously and in Firefox its after 'contextMenu' event.
+			// Adding longer timeout like 200ms still doesn't guarantee that test will pass on Firefox.
+			if ( CKEDITOR.env.gecko ) {
+				assert.ignore();
+			}
+
+			var range = new CKEDITOR.dom.range( editor.document ),
+				selectionRect;
+
+			range.setStart( editor.editable().getFirst(), 0 );
+
+			range.select();
+
+			selectionRect = range.getClientRects( true )[ 0 ];
+
+			editor.once( 'panelShow', function( evt ) {
+				setTimeout( function() {
+					resume();
```

Comment:
All assertions should be inside `resume` callback. Otherwise tests will pass even if assertions themselves fail:

<img width="593" alt="Tests passing even if assertions failed" src="https://user-images.githubusercontent.com/1078728/43004407-f905b834-8c2f-11e8-96a4-4c08e1f9b999.png">